Volume/Networking/Network.swift

```diff
@@ -87,3 +87,30 @@ struct GraphQLPublisher<Query: GraphQLQuery>: Publisher {
         subscriber.receive(subscription: subscription)
     }
 }
+
+class NetworkState: ObservableObject {
+    @Published var networkScreenFailed: [Screen : Bool] = [:]
+
+    public enum Screen: CaseIterable {
+        case homeList, publicationList, bookmarksList
+    }
+
+    init() {
+        for screen in Screen.allCases {
+            networkScreenFailed[screen] = false
+        }
+    }
+
+    func determineState(screen: Screen, _ completion: Subscribers.Completion<WrappedGraphQLError>) {
```

Comment:
This name is confusing. How about `handleCompletion` or `didCompleteRequest` or `requestDidComplete`?